cpp/src/parquet/statistics.cc

```diff
@@ -567,6 +568,27 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     SetMinMaxPair(comparator_->GetMinMax(values));
   }
 
+  void UpdateArrowDictionary(const ::arrow::Array& indices,
+                             const ::arrow::Array& dictionary) {
+    IncrementNullCount(indices.null_count());
+    IncrementNumValues(indices.length() - indices.null_count());
+
+    if (indices.null_count() == indices.length()) {
+      return;
+    }
+
+    ::arrow::compute::ExecContext ctx(pool_);
+    PARQUET_ASSIGN_OR_THROW(auto referenced_indices,
```

Comment:
please spell out the auto types here and below, its not really clear what the output type is.

Reply:
I removed all instances of `auto` unless it was blatantly obvious (e.g. returning from make_shared).  Although in this spot it is Datum which isn't much clearer :)